### PR TITLE
Add custom decoding for all message types

### PR DESCRIPTION
**Pull request by ericmillin**

### CODE REVIEW

File: fluent/protocol/transport.go [hunk 1]
```diff
@@ -139,103 +137,27 @@ type Entry struct {
...
-}
-
-// CompressedPackedForwardMode is just like PackedForwardMessage, except that
-// the msgpack byte stream containing the events/entries is compressed using
-// gzip.  The protocol spec states that the event stream may be formed by
-// concatenating multiple gzipped binary strings, but we do not claim to
-// support that yet.
-// Users of this type MUST set the option "compressed" => "gzip"
-//msgp:tuple CompressedPackedForwardMessage
-type CompressedPackedForwardMessage struct {
```
Comment (ericmillin) [inline fluent/protocol/transport.go:218]: We don't need CompressedPackedForwardMessage as it's just a PackedForwardMessage where the events are gzipped.

File: fluent/protocol/message_gen_test.go [hunk 2]
```diff
@@ -0,0 +1,3 @@
+package protocol
```
Comment (ericmillin) [inline fluent/protocol/message_gen_test.go:1]: need to stop this from getting generated

File: fluent/protocol/message_test.go [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+import (
+	"bytes"
+	"testing"
+
+	"github.com/tinylib/msgp/msgp"
+
+	. "github.com/IBM/fluent-forward-go/fluent/protocol"
+)
+
+func TestMarshalUnmarshalMessage(t *testing.T) {
```
Comment (ericmillin) [inline fluent/protocol/message_test.go:12]: I want to rewrite these using Ginkgo in another PR.

**Comment by Christopher-Wolf-ibm** [comment]
Everything looks good to me.

**Review by Christopher-Wolf-ibm — APPROVED** [review]

**Pull request merged**